scripts/irbrc.rb

```diff
@@ -73,3 +38,40 @@
 rescue => exception
   puts "Error : 'load #{ENV["HOME"]}/.irbrc' : #{exception.message}"
 end
+
+# Turn on completion.
+begin
+  require "readline"
+
+  require "irb/completion" rescue nil
+
+  # Turn on history saving.
+  # require "irb/ext/save-history"
+  # IRB.conf[:HISTORY_FILE] = File.join(ENV["HOME"], ".irb-history")
+
+  # Use an alternate way to on history saving until save-history is fixed.
+  #
+  #   bug:   http://redmine.ruby-lang.org/issues/show/1556
+  #   patch: http://pastie.org/513500
+  #
+  # This technique was adopted from /etc/irbrc on OS X.
+  histfile = File.expand_path(".irb-history", ENV["HOME"])
+  histfile = IRB.conf[:HISTORY_FILE] || histfile
```

Comment:
Sure. I'm not convinced that the readline code can be eliminated, but I'm happy to poke at it and let you know what I find. Obviously, it would be great if that whole chunk of code can be removed.